portality/view/publisher.py

```diff
@@ -54,7 +55,10 @@ def delete_application(application_id):
 
     # otherwise delegate to the application service to sort this out
     appService = DOAJ.applicationService()
-    appService.delete_application(application_id, current_user._get_current_object())
+    try:
+        appService.delete_application(application_id, current_user._get_current_object())
+    except NoSuchObjectException:
+        abort(400)
```

Comment:
Please update the response to return a 404 to ensure consistency with equivalent API requests.